ci-operator/step-registry/ovn/clusternetwork-cidr-expansion/ovn-clusternetwork-cidr-expansion-commands.sh

```diff
@@ -0,0 +1,107 @@
+    ready=$(echo "$machinesets" | /tmp/yq ".items[$i].status.readyReplicas")
+    available=$(echo "$machinesets" | /tmp/yq ".items[$i].status.availableReplicas")
+    name=$(echo "$machinesets" | /tmp/yq ".items[$i].metadata.name")
+    if [[ "$desired" != "$ready" || "$ready" != "$available" ]]; then
+        echo "Error: machine set $name has mismatched counts"
+        exit 1
+    fi
+    # Set NODE_TO_SCALE to the name of the first node in the list
+    if [[ $i == 0 ]]; then
+        NODE_TO_SCALE=$name
+        READY_COUNT=$ready
+    fi
+done
+
+echo "NODE_TO_SCALE=$NODE_TO_SCALE"
+oc scale --replicas=$(($READY_COUNT + 3)) machineset "$NODE_TO_SCALE" -n openshift-machine-api
+
+# wait for the two extra nodes to become ready, then validate that only 2 of the new nodes were allocated a subnet.
+# the 3rd extra node should be notReady and have no subnet because they are exhausted
+oc wait machinesets -n openshift-machine-api "$NODE_TO_SCALE" --for=jsonpath='{.status.readyReplicas}'=3 --timeout=1200s
+oc get nodes -o jsonpath='{range.items[*]} {.metadata.name} {"\t"} {.metadata.annotations.k8s\.ovn\.org/node-subnets} {"\n"}'
+node_subnet_assignments=$(oc get nodes -o jsonpath='{range.items[*]} {.metadata.name} {"\t"} {.metadata.annotations.k8s\.ovn\.org/node-subnets} {"\n"}')
+nodes_with_subnet=$(echo "$node_subnet_assignments" | grep -c "default.*10.128")
+nodes_without_subnet=$(( $(echo "$node_subnet_assignments" | grep -vc "default.*10.128") - 1 )) # -1 to account for blank line
+not_ready_node=$(oc get nodes | grep NotReady | cut -d ' ' -f1)
+oc describe node $not_ready_node
+oc describe node $not_ready_node | grep "nodeAdd: error adding node \"${not_ready_node}\": could not find \"k8s.ovn.org/node-subnets\" annotation"
+if [ $? -ne 0 ]; then
+    echo "Did not find a notReady node. Expected one node to be notReady because there are no subnets available"
+    exit 1
+fi
+
+# Check if there is exactly 1 node without a subnet and 8 nodes with a subnet
+if [ "$nodes_with_subnet" -ne 8 ] || [ "$nodes_without_subnet" -ne 1 ]; then
+  echo "failure: expected 8 nodes with subnets and 1 node with no subnet"
+  exit 1
+fi
+# patch the cluster to give it more ip space with /22
+oc patch Network.config.openshift.io cluster --type='merge' --patch '{ "spec":{ "clusterNetwork": [ {"cidr":"10.128.0.0/22","hostPrefix":26} ], "networkType": "OVNKubernetes" }}'
+
+# first wait for the network operator to move to Progressing=True
+if ! oc wait co network --for='condition=PROGRESSING=True' --timeout=120s; then
+  echo "the network operator never moved to Progressing=True. The clusterNetwork CIDR change may not have worked"
+  oc get co
+  exit 1
+fi
+
+wait_for_operators
+
+# check that 9th node became available. Should not have to wait long as it should have
+# moved to Ready state during the ovnk rollout process above
+oc wait machinesets -n openshift-machine-api "$NODE_TO_SCALE" --for=jsonpath='{.status.readyReplicas}'=4 --timeout=10s
+
```

Comment:
how about checking that all 9 nodes are now ready at this point?

Reply:
good one. done.